examples/functions/golang/grpc-serving/main.go

```diff
@@ -36,7 +48,88 @@ func main() {
+//              the players you want to assign to your DGS in Roster(s) when you call the 
+//              BackendAPI CreateAssignments() endpoint.  Might as well put them in rosters now.
+//  - error :   Use if you need to return an unrecoverable error.
+func makeMatches(ctx context.Context, logger *log.Entry, profile string, rosters []*pb.Roster, pools []*pb.PlayerPool) (string, []*pb.Roster, error) {
+
+	// Open Match will try to marshal your JSON roster to an array of protobuf Roster objects. It's 
+	// up to you if you want to fill these protobuf Roster objects or just write your Rosters in your 
+	// custom JSON blob. This example uses the protobuf Rosters.
+
+	// Used for tracking metrics.
+	var selectedPlayerCount int64
+
+	// Loop through all the team rosters sent in the call to create a match.
+	for ti, team := range rosters {
+		logger.Infof(" Attempting to fill team: %v", team.Name)
+
+		// Loop through all the players slots on this team roster.
+		for si, slot := range team.Players {
+
+			// Loop through all the pools and check if there is a pool with a matching name to the
+			// poolName for this player slot. Just one example of a way for your matchmaker to 
+			// specify which pools your MMF should search through to fill a given player slot. 
+			// Optional, feel free to change as you see fit.
+			for _, pool := range pools {
+				if slot.Pool == pool.Name && len(pool.Roster.Players) > 0 {
+
+					/////////////////////////////////////////////////////////
+					// These next few lines are where you would put your custom logic, such as 
+					// searching the pool for players with similar latencies or skill ratings 
+					// to the players you have already selected.  This example doesn't do anything
+					// but choose at random!
+					logger.Infof("Looking for player in pool: %v, size: %v", pool.Name, len(pool.Roster.Players))
+					randPlayerIndex := rand.New(rand.NewSource(time.Now().UnixNano())).Intn(len(pool.Roster.Players))
+
+					// Get random player with this index
+					selectedPlayer := pool.Roster.Players[randPlayerIndex]
+					logger.Infof("Selected player index %v: %v", randPlayerIndex, selectedPlayer.Id)
+
+					// Remove this player from the array as they are now used up.
+					pool.Roster.Players[randPlayerIndex] = pool.Roster.Players[0]
+
+					// This is a functional pop from a set.
+					_, pool.Roster.Players = pool.Roster.Players[0], pool.Roster.Players[1:]
+
+					// Write the player to the slot and loop.
+					rosters[ti].Players[si] = selectedPlayer
+					selectedPlayerCount++
+					break
+					/////////////////////////////////////////////////////////
+
+				} else {
+					// Weren't enough players left in the pool to fill all the
+					// slots.  If your game can handle non-full teams, feel
+					// free to customize this part as you see fit.
+					return "{\"error\": \"insufficient players\"}", rosters, errors.New("insufficient players")
```

Comment:
Why are we sending an error as JSON here? If anything the error text should be extracted from the error variable. It's go convention to return "" for string or nil for structs.

Reply:
Good catch. The harness itself handles it like how you stated - if there is an error, pulls error text out to return on the match error (does not use this string) - so fixed this! I ve updated the comment to indicate clearly that if error is returned in the error variable - then this field is ignored and should be "".